config_test.go

```diff
@@ -16,21 +16,45 @@ func TestValidateConfig(t *testing.T) {
 		t.Fatalf("TestValidateConfig: Config validation error exp(%v) failed(%v)", errNoConfigProvided, err)
 	}
 
-	// PSK and Certificate
+	// PSK and Certificate, valid cipher suites
 	cert, err := selfsign.GenerateSelfSigned()
 	if err != nil {
 		t.Fatalf("TestValidateConfig: Config validation error(%v), self signed certificate not generated", err)
 		return
 	}
 	config := &Config{
+		CipherSuites: []CipherSuiteID{TLS_PSK_WITH_AES_128_CCM_8, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256},
+		PSK: func(hint []byte) ([]byte, error) {
+			return nil, nil
+		},
+		Certificates: []tls.Certificate{cert},
+	}
+	if err = validateConfig(config); err != nil {
+		t.Fatalf("TestValidateConfig: Client error exp(%v) failed(%v)", nil, err)
```

Comment:
This seems a bit odd. The first `%v` will always be nil?

Reply:
The convention in most Go project, as well as this code base is to use `expected: something, got: something else`. Make it easy for humans to read, not for computers to parse.

Reply:
Would hard-coding `nil` in the string be preferable? I had copied the format from other tests, and it feels to me that keeping both parameters as interpolated into the string would be consistent with reading other tests (i.e. you can always look at the last two parameters for expected and got when reading the tests source), but I'm okay with whatever you feel would be best for the project overall.

Reply:
Yeah, lets just keep it for now as it is. I was reading through the change and somewhat surprised, until I opened the whole file and realised we do this everywhere for these tests. I'll go in at some point and fix that when the urge strikes.